### PR TITLE
Making Fused gate matrix backend dependent

### PR DESCRIPTION
**Pull request by BrunoLiegiBastonLiegi**
This just makes the fused gate to be constructed through the backend and avoid converting to numpy. The reason for this is that in some cases conversion to numpy can be tricky: with gpu backends you have to copy data to cpu, whereas for ml backends conversion to numpy is not always possible and will fail (e.g. pytorch with gradients attached, symbolic tensorflow or traced jax arrays).

Checklist:
- [ ] Reviewers confirm new code works as expected.
- [ ] Tests are passing.
- [ ] Coverage does not decrease.
- [ ] Documentation is updated.

### CODE REVIEW

**Comment by BrunoLiegiBastonLiegi** [comment]
This is no longer needed, I ended up directly overwriting the method in the qiboml backends qiboteam/qiboml#89.
